### ISSUE TITLE
Switching projects can break build in status-bar github integration

### ISSUE THREAD

**Issue by Laserzwei** [jccguimaraes/atom-project-viewer#174]
# Issue
The build in github commit button (status bar) breaks, when a project is switched within the window (**works fine with setting "Open in new Window" active**).
broken Button: 
![](https://imgur.com/4MsgLpB.jpg)

after reloading window (ctrl-shift-F5):
![](https://imgur.com/gG4o2lC.jpg)
This affects the pull/push button next to it as well.

## Environment
- **OS**: Win7 64bit;
- **Atom**: 1.21.1 x64;

## Steps

1. Have setting "Open in new Window" inactive.
2. Have a project using github.
3. Change something.
4. Switch project without commiting.
5. Return to 1st project.

can be resolved by adding the "github" **after** the status-bar package to the List of packages to reload.
However this would causes the following error:
Error: Emitter has been disposed

    at Emitter.module.exports.Emitter.on (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\event-kit\lib\emitter.js:77:21)
    at Repository.onDidUpdate (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\github\lib\models\repository.js:149:31)
    at ModelObserver.setActiveModel (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\github\lib\models\model-observer.js:35:58)
    at Wrapper.componentWillMount (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\github\lib\decorators\observe-model.js:82:34)
    at ReactCompositeComponentWrapper.performInitialMount (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:348:20)
    at ReactCompositeComponentWrapper.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:255:27)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at ReactDOMComponent.mountChildren (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMultiChild.js:234:50)
    at ReactDOMComponent._createInitialChildren (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactDOMComponent.js:701:38)
    at ReactDOMComponent.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactDOMComponent.js:520:18)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at ReactCompositeComponentWrapper.performInitialMount (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:368:40)
    at ReactCompositeComponentWrapper.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:255:27)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at ReactDOMComponent.mountChildren (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMultiChild.js:234:50)
    at ReactDOMComponent._createInitialChildren (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactDOMComponent.js:701:38)
    at ReactDOMComponent.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactDOMComponent.js:520:18)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at ReactCompositeComponentWrapper.performInitialMount (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:368:40)
    at ReactCompositeComponentWrapper.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:255:27)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at ReactCompositeComponentWrapper.performInitialMount (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:368:40)
    at ReactCompositeComponentWrapper.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactCompositeComponent.js:255:27)
    at Object.mountComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactReconciler.js:43:41)
    at mountComponentIntoNode (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMount.js:102:38)
    at ReactReconcileTransaction.perform (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\Transaction.js:141:26)
    at batchedMountComponentIntoNode (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMount.js:124:21)
    at ReactDefaultBatchingStrategyTransaction.perform (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\Transaction.js:141:26)
    at Object.batchedUpdates (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactDefaultBatchingStrategy.js:60:32)
    at Object.batchedUpdates (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactUpdates.js:95:33)
    at Object._renderNewRootComponent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMount.js:317:24)
    at Object._renderSubtreeIntoContainer (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMount.js:399:38)
    at Object.render (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\react-dom\lib\ReactMount.js:420:29)
    at GithubPackage.rerender (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\github\lib\github-package.js:336:30)
    at GithubPackage.consumeStatusBar (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\github\lib\github-package.js:383:16)
    at Provider.module.exports.Provider.provide (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\service-hub\lib\provider.js:34:58)
    at ServiceHub.module.exports.ServiceHub.provide (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\service-hub\lib\service-hub.js:30:26)
    at Package.module.exports.Package.activateServices (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:446:77)
    at Package.module.exports.Package.activateNow (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:256:22)
    at C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:225:38
    at Package.module.exports.Package.measure (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:99:21)
    at C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:218:32
    at Package.module.exports.Package.activate (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:215:40)
    at PackageManager.activatePackage (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package-manager.js:690:42)
    at packagesToEnable.forEach.name (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package-manager.js:439:51)
    at Array.forEach (native)
    at disabledPackagesSubscription.config.onDidChange (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package-manager.js:439:30)
    at C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\config.js:693:26
    at Function.module.exports.Emitter.simpleDispatch (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\event-kit\lib\emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\node_modules\event-kit\lib\emitter.js:141:34)
    at Config.module.exports.Config.emitChangeEvent (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\config.js:884:35)
    at Config.module.exports.Config.setRawValue (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\config.js:670:25)
    at Config.module.exports.Config.set (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\config.js:248:20)
    at Config.module.exports.Config.removeAtKeyPath (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\config.js:407:18)
    at Package.module.exports.Package.enable (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package.js:84:32)
    at PackageManager.enablePackage (C:\Users\MB\AppData\Local\atom\app-1.21.1\resources\app\src\package-manager.js:245:18)
    at list.forEach.pkg (C:\Users\MB\.atom\packages\project-viewer\src\packages.js:16:42)
    at Array.forEach (native)
    at Object._enablePackages [as enable] (C:\Users\MB\.atom\packages\project-viewer\src\packages.js:14:8)
    at HTMLLIElement._openOnWorkspace [as openOnWorkspace] (C:\Users\MB\.atom\packages\project-viewer\src\project-view.js:329:24)
    at HTMLLIElement._onClickEvent (C:\Users\MB\.atom\packages\project-viewer\src\project-view.js:17:8)

Edit: This issue is related to:
#155
Removing status-bar (and github) from the reload list solved this without an error

You made a very useful package!

**Comment by jccguimaraes** [comment]
hi @Laserzwei,

Thanks for reporting this.

Will try to make an in depth look at this.

Sorry for this.

Also thanks for the vote of confidence

**Comment by maximkrusina** [comment]
Hi, have the same issue here, but in my case, I'm not using GitHub but general Git repo.

**Comment by maximkrusina** [comment]
PS: workaround: switch to another project and back.

**Comment by jccguimaraes** [comment]
hey @Laserzwei and @maximkrusina,

Been trying to get a workaround / fix for this, but it's getting really hard due to the nature of this feature. Just as a reminder, Atom does not provide context switch between projects in the same instance without a reload mechanism. So I have to consider messing around with the plugins, `tree-view`, `status-bar`, `find-and-replace` and there is no win win solution at the moment. But i will try to dig more as plans for the next big release are shaping up.

**Comment by jccguimaraes** [comment]
@Laserzwei can you check this with new version?

**Comment by maximkrusina** [comment]
Yep, I just opened Atom after a few months, nice coincidence :) Going to test it, coz franky, your plugin is the best, I didn't like all the others :)

**Comment by Laserzwei** [comment]
I've mentioned in the first post that removing status-bar and github from the "List of Packages to reload" config-list, resolved this issue for me. Hence I've been running your package without it and without any related issue (#188 bugs me a little, but I posted workarounds as of now). 
To text your changes: I added both back to the reload list (in 1.3.0) and the same problem reoccured. The github integration becomes not clickable, but no error is prompted.


Question: What are the default packages that get reloaded?
